docs/dev/notebooks.md

```diff
@@ -0,0 +1,55 @@
+# Notebooks guidelines 
+
+Our guides and tutorials are frequently written using iPython Notebooks. The notebooks require specific formatting, are continuosly tested (when possible) and we have a specific process to manage the lifecycle of a notebook before and after a Cirq release.    
+
+## Formatting 
+
+Formatting is easy, the script `check/nbformat` should tell you if your notebooks are formatted or not.
+You can apply the changes in one go with `check/nbformat --apply`. It is recommended to add this to you [git pre-commit hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks), to save feedback time and CI resources. 
+
+## Output cells 
+
+Output cells typically should not be saved in the notebook. They will be generated for the final site.
+The exception to this rule are notebooks with external dependencies (see below). 
+
+## Header
+
+We also expect a standard header to be included in all of our notebooks: 
+- the links to colab, github and the main site (quantumai.google/cirq)
+- optional package installation (you can assume Colab dependencies exist)
+ 
+
+## Testing 
+
+Those notebooks that don't have any external dependencies (e.g. authentication) can be executed in an isolated environment are being tested on a continuous basis. 
+See the `dev_tools/notebooks` directory for the two tests: 
+- notebook_tests.py - to test notebooks against the current branch
+- isolated_notebook_tests.py - to test notebooks against the latest released version of Cirq
+
+## Notebooks with external dependencies 
+
+Unfortunately we have no way to test notebooks yet easily with external API dependencies, e.g. cirq.google's Engine API. 
+These notebooks should be excluded from both tests. 
+
+The site that generates the outputs for notebooks also can't handle external dependencies. 
+
+Thus, for notebooks with external dependencies, **all cells must have their outputs saved in the notebook file**. This ensures that the site pipeline will skip these notebooks.  
+
+## Lifecycle 
+
+Notebooks are handled differently based on whether they rely on features in the pre-release build of cirq or not. 
```

Comment:
"Notebooks are handled..." gives the impression that the behavior below is applied automatically. We should make it clear that the notebook author is responsible for these steps.